src/_components/AssetDetails/AssetDetailsContainer.jsx

```diff
@@ -3,29 +3,27 @@ import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 import { isEmpty } from 'lodash';
 import { Container, Divider, Header } from 'semantic-ui-react';
-import { getAssetDetail } from '../_actions/asset.actions';
-import { loadAssetAssigneeUsers } from '../_actions/users.actions';
-import AssetDetailContent from './AssetDetailContent';
-import NavBarComponent from '../_components/NavBarContainer';
-import LoaderComponent from './LoaderComponent';
+import { getAssetDetail } from '../../_actions/asset.actions';
+import { loadAssetAssigneeUsers } from '../../_actions/users.actions';
+import AssetDetailsComponent from '../../components/AssetDetails/AssetDetailsComponent';
+import NavBarComponent from '../NavBarContainer';
+import LoaderComponent from '../../components/LoaderComponent';
 
-export class AssetDetailComponent extends Component {
+export class AssetDetailsContainer extends Component {
   state = {
     assignedUser: {},
     hasError: this.props.hasError,
     errorMessage: this.props.errorMessage
   };
 
   componentDidMount() {
-    const { assetDetail, assetAsigneeUsers, match } = this.props;
-    if (isEmpty(assetDetail)) {
-      const { id } = match.params;
-      this.props.getAssetDetail(id);
-    }
-    if (isEmpty(assetAsigneeUsers)) {
-      this.props.loadAssetAssigneeUsers();
-    }
+    const { match } = this.props;
+    const { id } = match.params;
+
+    this.props.getAssetDetail(id);
```

Comment:
This will make an API call all the time, even if the `assetDetail` exist in `props.location.state`. This is what `if (isEmpty(assetDetail))` was trying to avoid as it negatively impacts performance and UX. The best way forward for this is to **"confirm that the data in the store is for a different id, and there's no location state"**. This should be done in `mapStateToProps` and you just pass a prop to the component, let's say `shouldFetchDetails`.  That is:
```jsx
if (shouldFetchDetails) {
  this.props.getAssetDetail(id);
}
```
Then once you have created a container as per my previous comment, you can have the following in it: 
```jsx
import get from 'lodash/get';
import isEmpty from 'lodash/isEmpty';

// ensure you are comparing the correct ids. I have used `assetDetail.id` for demonstration purpose
const isSameId = (assetDetail, props) => (assetDetail.id === props.match.params.id);

// If same id, then the data exist in store. Otherwise, check for it in props.location.state
const getAssetDetail = (assetDetail, props) => {
  const hasSameId = isSameId(assetDetail, props);
  if (hasSameId) {
    return assetDetail;
  }
  return get(props.location, 'state', {});
}

const mapStateToProps = ({ asset, usersList }, ownProps) => {
  const { assetDetail } = asset;
  const details = getAssetDetail(assetDetail, ownProps);

 // note that details represents `assetDetails` passed as props. I wanted to make the distinction clear
  return {
    shouldFetchDetails: isEmpty(details),
    details,
    ...
    buttonLoading
  };
}
```
Also, correct me if I'm wrong but my understanding of `loadAssetAssigneeUsers` is that it fetches a list of all users  that can be assigned an asset, and it is not specific to a particular asset. Therefore, we should only fetch the users when there is no data in store. This means that we need to check for it hence the previous logic is what we need. i.e.
```jsx
if (isEmpty(assetAsigneeUsers)) {
  this.props.loadAssetAssigneeUsers();
}
```